sys/include/net/credman.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_credman (D)TLS Credential Manager
+ * @ingroup     net
+ * @brief       Credentials management module for (D)TLS
+ *
+ * @{
+ *
+ * @file
+ * @brief       (D)TLS credentials management module definitions
+ *
+ * @note        This module DOES NOT copies the credentials into the system. It
+ *              just holds the pointers to the credentials given by the user.
+ *              The user must make sure that these pointers are valid during the
+ *              lifetime of the application.
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#ifndef NET_CREDMAN_H
+#define NET_CREDMAN_H
+
+#include <unistd.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Maximum number of credentials in credential pool
+ */
+#ifndef CREDMAN_MAX_CREDENTIALS
+#define CREDMAN_MAX_CREDENTIALS  (2)
+#endif
+
+/**
+ * @brief Buffer of the credential
+ */
+typedef struct {
+    void *s;                /**< Pointer to the buffer */
+    size_t len;             /**< Length of s */
+} credman_buffer_t;
+
+/**
+ * @brief PSK parameters
+ */
+typedef struct {
+    credman_buffer_t key;   /**< Key buffer */
+    credman_buffer_t id;    /**< Id buffer */
+    credman_buffer_t hint;  /**< Hint buffer */
+} psk_params_t;
+
+/**
+ * @brief ECDSA public keys
+ */
+typedef struct {
+    const void *x;          /**< X part of the public key */
+    const void *y;          /**< Y part of the public key */
+} ecdsa_public_key_t;
+
+/**
+ * @brief ECDSA parameters
+ */
+typedef struct {
+    const void * private_key;           /**< Pointer to the private key */
+    ecdsa_public_key_t public_key;      /**< Public key */
+    ecdsa_public_key_t *client_keys;    /**< Array of clients public keys */
+    size_t client_keys_size;            /**< Size of clients_keys */
+} ecdsa_params_t;
+
+/**
+ * @brief Tag of the credential.
+ */
+typedef uint16_t credman_tag_t;
+
+/**
+ * @brief Used to identify credentials for application libraries.
+ */
+#define CREDMAN_TAG_GLOBAL  (999)
+
+/**
+ * @brief Used to signal empty/no tag
+ */
+#define CREDMAN_TAG_EMPTY   (0)
+
+/**
+ * @brief Credential types
+ */
+typedef enum {
+    CREDMAN_TYPE_EMPTY  = 0,
+    CREDMAN_TYPE_PSK    = 1,
+    CREDMAN_TYPE_ECDSA  = 2,
+} credman_type_t;
+
+/**
+ * @brief Credential informations
```

Comment:
```suggestion
 * @brief Credential information
```

(the plural of "information" is "information")